### ISSUE TITLE
[assemble] Support assembling `.deb` packages

### ISSUE THREAD

**Issue by aalmiray** [jreleaser/jreleaser#1240]
While the `jpackage` assembler may be used to create such files the input must be a Java "program".
[jdeb](https://github.com/tcurdt/jdeb) may be used to create `.deb` files for non Java binaries.

**Comment by aalmiray** [comment]
Turns out `jdeb` may not be suitable as some utility classes still depend on Ant, the control file is parsed one more time before data files are added to it. Might need to implement our own `.deb` assembler 😥

**Comment by aalmiray** [comment]
More details on control files -> https://www.debian.org/doc/debian-policy/ch-controlfields.html

```
$ ar -t jreleaser-installer_1.10.0-1_amd64.deb 
debian-binary
control.tar.zst
data.tar.zst

$ ar -x jreleaser-installer_1.10.0-1_amd64.deb 
$ cat debian-binary 
2.0

$ tar tvf control.tar.zst 
drwxr-xr-x  0 root   root        0 Dec 31 16:41 ./
-rw-r--r--  0 root   root      279 Dec 31 16:41 ./control
-rwxr-xr-x  0 root   root      913 Dec 31 16:41 ./postinst
-rwxr-xr-x  0 root   root      814 Dec 31 16:41 ./postrm
-rwxr-xr-x  0 root   root      575 Dec 31 16:41 ./preinst
-rwxr-xr-x  0 root   root      839 Dec 31 16:41 ./prerm

$ tar tvf data.tar.zst 
drwxr-xr-x  0 root   root        0 Dec 31 16:41 ./
drwxr-xr-x  0 root   root        0 Dec 31 16:41 ./opt/
drwxr-xr-x  0 root   root        0 Dec 31 16:41 ./opt/jreleaser/
...

$ cat control
Package: jreleaser
Version: 1.10.0-1
Section: misc
Maintainer: JReleaser <jreleaser@kordamp.org>
Priority: optional
Architecture: amd64
Provides: jreleaser
Description: Release projects quickly and easily with JReleaser
Depends: libc6, xdg-utils, zlib1g 
Installed-Size: 107364
```

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.16.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.16.0))
